### PR TITLE
Enhanced malware corpus & update file names to match behaviors

### PR DESCRIPTION
**Pull request by ryantxu1**

### CODE REVIEW

File: anti-behavioral-analysis/debugger-detection.md [hunk 1]
```diff
@@ -18,11 +18,11 @@ Methods
...
+|**Check Processes**|B0001.038|The malware may check running processes for specific strings such as "malw" to detect a analysis environment.|
 |**CloseHandle**|B0001.003|(NtClose); If an invalid handle is passed to the CloseHandle function and a debugger is present, then an EXCEPTION_INVALID_HANDLE (0xC0000008) exception will be raised. [[7]](#7)|
 |**Debugger Artifacts**|B0001.004|Malware may detect a debugger by its artifact (window title, device driver, exports, etc.).|
 |**Hardware Breakpoints**|B0001.005|(SEH/GetThreadContext); Debug registers will indicate the presence of a debugger. See [[7]](#7) for details.|
-|**Interrupt 0x2d**|B0001.006|If int 0x2d is mishandled by the debugger, it can cause a single-byte instruction to be inadvertently skipped, which can be detected by malware.|
-|**Interrupt 1**|B0001.007|[[7]](#7)|
+|**Interruption**|B0001.006|If an interruption is mishandled by the debugger, it can cause a single-byte instruction to be inadvertently skipped, which can be detected by malware.|
```
Comment (dzbeck) [inline anti-behavioral-analysis/debugger-detection.md:25]: Please add another sentence, something like, "Examples include Interrupt 0x2d and Interrupt 1 [[7]](#7)." (check to see if [7] ref is appropriate)

File: defense-evasion/process-injection.md [hunk 2]
```diff
@@ -40,15 +40,27 @@ Malware Examples
...
 |[**WebCobra**](../xample-malware/webcobra.md)|2018|Injects minor code into a running process.|
+|[**CryptoWall**](../xample-malware/cryptowall.md)|2014| [[6]](#6)|
+|[**Hupigon**](../xample-malware/hupigon.md)|2013|The malware injects itself into processes such as cmd.exe, notepad.exe [[7]](#7)|
+|[**BlackEnergy**](../xample-malware/blackenergy.md)|2007|Bypasses UAC using a Shim Database instructing SndVol.exe to execute cmd.exe instead, allowing for elevated execution  [[8]](#8)|
+|[**Stuxnet**](../xample-malware/stuxnet.md)|2010|Stuxnet injects the entire DLL into another process and then just calls the particular export  [[9]](#9)|
 
 References
 ----------
-<a name="1">[1]</a> Ashkan Hosseini, *Ten Process Injection Techniques: A Technical Survey of Common and Trending Process Injection Techniques*, July 2017. https://www.elastic.co/blog/ten-process-injection-techniques-technical-survey-common-and-trending-process
+<a name="1">[1]</a> Ashkan Hosseini, *Ten Process Injection Techniques: A Technical Survey of Common and Trending Process Injection Techniques*, July 2017. https://www.elastic.co/blog/ten-process-injectionion-techniques-technical-survey-common-and-trending-process
```
Comment (dzbeck) [inline defense-evasion/process-injection.md:50]: This link looks wrong. "-injectionion" should be "-injection"

File: defense-evasion/process-injection.md [hunk 3]
```diff
@@ -40,15 +40,27 @@ Malware Examples
...
 
 <a name="2">[2]</a> https://www.cyber.nj.gov/threat-profiles/trojan-variants/poison-ivy
 
 <a name="3">[3]</a> https://github.com/LordNoteworthy/al-khaser
 
 <a name="4">[4]</a> https://citizenlab.ca/2016/04/between-hong-kong-and-burma/
 
-<a name="5">[5]</a> https://www.ired.team/offensive-security/code-injection-process-injection/executing-shellcode-with-createfiber
+<a name="5">[5]</a> https://www.ired.team/offensive-security/code-injection-process-injectionion/executing-shellcode-with-createfiber
```
Comment (dzbeck) [inline defense-evasion/process-injection.md:58]: remove extra "ion" in link.

File: micro-behaviors/cryptography/hashed-message-authentication-code.md [hunk 4]
```diff
@@ -7,4 +7,4 @@
 
 Hashed Message Authentication Code
 ==================================
-Malware uses an HMAC schema.
\ No newline at end of file
+Malware uses an hashed-message-authentication-code schema.
```
Comment (dzbeck) [inline micro-behaviors/cryptography/hashed-message-authentication-code.md:10]: add acronym, change "an" to "a", and remove hyphens: "Malware uses a hashed message authentication code (HMAC) schema."

File: xample-malware/gotbotkr.md [hunk 5]
```diff
@@ -13,12 +13,46 @@ Modified version of a publicly available backdoor name GoBot2. Modifications are
...
+|[Lateral Movement::Replication Through Removable Media (T1091)](https://attack.mitre.org/techniques/T1091/)|GoBotKR can drop itself onto removable media and relies on Autorun to execute the malicious file when a user opens the removable media on another system. [[1]](#1)|
+|[Command and Control::Connection Proxy (T1090)](https://attack.mitre.org/techniques/T1090/)|GoBotKR can be used as a proxy server. [[1]](#1)|
+|[Command and Control::Data Encoding (T1132)](https://attack.mitre.org/techniques/T1132/)|The communication with the C&C server is base64 encoded. [[1]](#1)|
+|[Command and Control::Remote File Copy (T1105)](https://attack.mitre.org/techniques/T1105/)|GoBotKR can download additional files and update itself. [[1]](#1)|
+|[Command and Control::Standard Application Layer Protocol (T1071)](https://attack.mitre.org/techniques/T1071/)|GoBotKR uses HTTP or HTTPS for C&C. [[1]](#1)|
+|[Command and Control::Uncommonly Used Port (T1571)](https://attack.mitre.org/techniques/T1571/)|GoBotKR uses non-standard ports, such as 6446, 6556 and 7777, for C&C. [[1]](#1)|
+
 Behaviors
```
Comment (dzbeck) [inline xample-malware/gotbotkr.md:40]: As a counterpart to "ATT&CK Techniques", these sections for all examples should be labeled "MBC Behaviors" - to make it clear that the distinction is MBC vs. ATT&CK.

File: xample-malware/gotbotkr.md [hunk 6]
```diff
@@ -13,12 +13,46 @@ Modified version of a publicly available backdoor name GoBot2. Modifications are
...
 ---------
 |Name|Use|
 |---|---|
-|[Process Discovery](https://attack.mitre.org/techniques/T1057/)|GotBotKR monitors whether the first instance is still active. [[1]](#1)|
-|[Install Additional Program](../execution/install-prog.md)|GotBotKR reinstalls its running instance if it is removed. [[1]](#1)|
+|[Execution::Install Additional Program](../execution/install-additional-program.md)|GotBotKR reinstalls its running instance if it is removed. [[1]](#1)|
```
Comment (dzbeck) [inline xample-malware/gotbotkr.md:44]: add ID (B0023)

File: xample-malware/hupigon.md [hunk 7]
```diff
@@ -11,12 +11,26 @@ Hupigon
...
+|Name|Use|
+|---|---|
+|[Defense Evasion::Obfuscated Files or Information:Software Packing (T1027.002)](https://attack.mitre.org/techniques/T1027/002/)|The malware may be packed with UPX [[2]](#2)|
+
 Behaviors
 ---------
 |Name|Use|
 |---|---|
-|[Registry Run Keys / Startup Folder](../persistence/registry-run-startup.md)|Hupigon drops the file "Systen.dll" and adds the registry entry: HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Winlogon\Notify\BITS DllName = "%System%\Systen.dll". [[1]](#1)|
+|[Registry Run Keys / Startup Folder](../persistence/registry-run-keys-startup-folder.md)|Hupigon drops the file "Systen.dll" and adds the registry entry: HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows NT\CurrentVersion\Winlogon\Notify\BITS DllName = "%System%\Systen.dll". [[1]](#1)|
```
Comment (dzbeck) [inline xample-malware/hupigon.md:25]: add ID (F0012)

File: xample-malware/kraken.md [hunk 8]
```diff
@@ -11,12 +11,19 @@ Kraken
...
+|[Command and Control::Dynamic Resolution:Domain Generation Algorithms (T1568.002)](https://attack.mitre.org/techniques/T1568/002/)|Uses a domain name generator.  [[1]](#1)|
+|[Command and Control::Application Layer Protocol::Web Protocols (T1071.001)](https://attack.mitre.org/techniques/T1071/001/)|The malware uses HTTP to communicate with C2 [[1]](#1)|
+
 Behaviors
 ---------
 |Name|Use|
 |---|---|
-|[Domain Name Generation](../command-and-control/domain-name-generate.md)|Uses a domain name generator. [[1]](#1)|
-|[Memory Dump Evasion](../anti-behavioral-analysis/evade-memory-dump.md)|Dumping Kraken's c.dll module from the heap of its own process is tricky because its PE-header is wiped out. [[1]](#1)|
+|[Command and Control::Domain Name Generation (B0003)](../command-and-control/domain-name-generation.md)| [[1]](#1)|
```
Comment (dzbeck) [inline xample-malware/kraken.md:25]: was the description removed? ("Uses a domain name generator. [[1]](#1)")

File: xample-malware/mebromi.md [hunk 9]
```diff
@@ -4,19 +4,27 @@
...
 Mebromi
 =======
 A BIOS bootkit.
 
+
+ATT&CK Techniques
+-----------------
+|Name|Use|
+|---|---|
+|[Defense Evasion::Pre-OS Boot::System Firmware (T1542.001)](https://attack.mitre.org/techniques/T1542/001/)|Malware performs BIOS modification and can download and execute a file as well as protect itself from removal [[1]](#1)|
```
Comment (dzbeck) [inline xample-malware/mebromi.md:19]: When malware is in ATT&CK, only techniques that have been enhanced in MBC (or techniques that were missed in ATT&CK) should be in the list under "ATT&CK techniques". There should also be a statement like, "Please see [Mebromi](https://attack.mitre.org/software/S0001/) for a complete list of ATT&CK techniques" (maybe in the last line of the table?)

File: xample-malware/trickbot.md [hunk 10]
```diff
@@ -11,8 +11,9 @@ TrickBot
 ========
 Trojan spyware program that has mainly been used for targeting banking sites. TrickBot is written in the C++ programming language.
 
-Behaviors
----------
+
+ATT&CK Techniques
```
Comment (dzbeck) [inline xample-malware/trickbot.md:15]: Trickbot is in ATT&CK - are all the techniques listed new to the analysis (not in ATT&CK)?

File: xample-malware/webcobra.md [hunk 11]
```diff
@@ -17,22 +17,22 @@ Behaviors
 |---|---|
 |[File and Directory Discovery](https://attack.mitre.org/techniques/T1083/)|[[1]](#1)|
 |[Query Registry](https://attack.mitre.org/techniques/T1012/)|[[1]](#1)|
-|[System Information Discovery](../discovery/system-info-discover.md)|Learns about the system so it can drop compatible miner software.  [[1]](#1)|
+|[System Information Discovery](../discovery/system-information-discovery.md)|Learns about the system so it can drop compatible miner software.  [[1]](#1)|
```
Comment (dzbeck) [inline xample-malware/webcobra.md:20]: These and other techniques need IDs.

File: xample-malware/blackenergy.md [hunk 12]
```diff
@@ -15,8 +15,41 @@ Behaviors
 ---------
 |Name|Use|
 |---|---|
-|[Denial of Service](../impact/denial-of-service.md)|Launches distributed denial of service attacks that can target more than one IP address per hostname. [[1]](#1)|
+|[Impact::Denial of Service (B0033)](../impact/denial-of-service.md)|Originally built to launch distributed denial of service attacks that can target more than one IP address per hostname  [[1]](#1)|
+|[Defense Evasion::Process Injection::Injection using Shims (E1055.m05)](../defense-evasion/process-injection.md)|Bypasses UAC using a Shim Database instructing SndVol.exe to execute cmd.exe instead, allowing for elevated execution  [[1]](#1)|
```
Comment (dzbeck) [inline xample-malware/blackenergy.md:19]: There is corresponding software in ATT&CK, so under an "ATT&CK Techniques" heading should be a pointer to the ATT&CK link. This list should be labeled "MBC Behaviors."

File: xample-malware/mebromi.md (hunk 9 above)
Comment (ryantxu1) [reply xample-malware/mebromi.md:19]: Yup will do this next

File: xample-malware/trickbot.md (hunk 10 above)
Comment (ryantxu1) [reply xample-malware/trickbot.md:15]: Ah no this was present in the file beforehand, will be changing it according to new protocol

File: xample-malware/bagle.md [hunk 13]
```diff
@@ -11,7 +11,7 @@ Bagle
 =====
 A mass-mailing computer worm affecting Microsoft Windows. [[1]](#1)
 
-Behaviors
+MBC MBC Behaviors
```
Comment (dzbeck) [inline xample-malware/bagle.md:14]: is "MBC" here twice?

**Review by dzbeck — APPROVED** [review]

**Pull request merged**